cou/steps/hypervisor.py

```diff
@@ -29,23 +31,25 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass(frozen=True)
-class HypervisorMachine:
-    """Hypervisor machine containing units for multiple applications."""
-
-    name: str
-    units: list[COUUnit]
-
-
-@dataclass(frozen=True)
+@dataclass
 class HypervisorGroup:
     """Group of hypervisors.
 
     For example, this can represent a group of all hypervisors in a single availability zone.
     """
 
     name: str
-    hypervisors: list[HypervisorMachine]
+    apps: dict[str, list[COUUnit]]
+
+    def __eq__(self, other: Any) -> bool:
+        """Equal magic method for HypervisorGroup.
+
+        :param other: HypervisorGroup object to compare.
+        :type other: any
+        :return: True if equal False if different.
+        :rtype: bool
+        """
+        return other.name == self.name
```

Comment:
If we are accepting `Any` type for comparison, we should consider the case where that object doesn't have `name` attribute. The current code would raise un-catched error.

Reply:
Good point, I copy and paste this from other class, so I will fix it there also.